src/content.py

```diff
@@ -0,0 +1,111 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Module for checking for 3-way merge conflicts and create content based on a 3 way merge."""
+
+import difflib
+import tempfile
+from pathlib import Path
+
+from git.exc import GitCommandError
+from git.repo import Repo
+
+from .exceptions import ContentError
+
+
+def conflicts(base: str, theirs: str, ours: str) -> str | None:
+    """Check for merge conflicts based on the git merge algorithm.
+
+    Args:
+        base: The starting point for both changes.
+        theirs: The other change.
+        ours: The local change.
+
+    Returns:
+        The description of the merge conflicts or None if there are no conflicts.
+    """
+    try:
+        merge(base=base, theirs=theirs, ours=ours)
+    except ContentError as exc:
+        return str(exc)
+    return None
```

Comment:
Whilst the logic is relatively simple, the intent of the method is different. the `merge` method is basically intended to be used to merge content that shouldn't have merge conflicts (as used by the `actions` module). This function is intended to let the caller get the conflicts, if any, where a conflict isn't an error. It means that the interface is better for the caller.

If only the `merge` method exists, the caller that just wants to check for conflicts would need to implement exception handling and know that the information about the conflict is in the exception message rather than just being able to ask "what are the conflicts with this" which is a much cleaner interface.

Any thoughts?

Reply:
We it's a bit similar case than the iterator. There's no other usage of the "merge" method that this one, therefore it's more "for future use" and I tend to avoid the "for future use" because the dev that will make the change doesn't necessarily understands the intention behind the choices and could make the change differently.
That more a "philosophy" in such cases I tend to merge the methods and if some refactoring is needed by a future change we'll do it at that time. I apply that philosophy unless I know that a shortly coming change will make use of this method.

Reply:
There is actually use of `conflicts` in the `check.py` module and `merge` is used in `actions.py`

Reply:
Oh I missed it, sorry the PR is pretty big ^^;